redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -335,5 +337,304 @@ inline void
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const std::vector<std::string>* endpoints =
+                std::get_if<std::vector<std::string>>(&(vEndpoints));
+            if (endpoints == nullptr)
+            {
+                BMCWEB_LOG_ERROR << "Failed to get Associations endpoints "
+                                    "for the given object path: "
+                                 << resourceObjPath.str;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            bool found = false;
+            std::string hwStatusEventObj;
+            for (const auto& endpoint : *endpoints)
+            {
+                if (sdbusplus::message::object_path(endpoint)
+                        .parent_path()
+                        .filename() == "hw_isolation_status")
+                {
+                    hwStatusEventObj = endpoint;
+                    found = true;
+                    break;
+                }
+            }
+
+            if (!found)
+            {
+                // No event so the respective hardware status doesn't need
+                // any Redfish status condition
+                return;
+            }
+
+            // Get the dbus service name of the hardware status event object
+            crow::connections::systemBus->async_method_call(
+                [aResp, hwStatusEventObj](const boost::system::error_code ec,
+                                          const GetObjectType& objType) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "DBUS response error [" << ec.value() << " : "
+                            << ec.message()
+                            << "] when tried to get the dbus name "
+                            << "of the hardware status event object "
+                            << hwStatusEventObj;
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    if (objType.size() > 1)
+                    {
+                        BMCWEB_LOG_ERROR << "More than one dbus service "
+                                         << "implemented the same hardware "
+                                         << "status event object "
+                                         << hwStatusEventObj;
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    if (objType[0].first.empty())
+                    {
+                        BMCWEB_LOG_ERROR << "The retrieved hardware status "
+                                         << "event object dbus name is empty";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    using AssociationsValType = std::vector<
```

Comment:
Don't we already have this defined somewhere you can use ?

Reply:
E.g. see https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/47776/8/redfish-core/lib/systems.hpp

Reply:
> Do you want to change that upstream and use it here?

Sure if you don't mind

Reply:
@gtmills I just looked to add in the upstream but, looks nowhere using the "**Associations**" property and using only **endpoints** in the upstream. 

downstream code:
```
-bash-4.2$ git grep -in "\"Associations\""
redfish-core/include/utils/hw_isolation.hpp:523:                                if (property.first == "Associations")
redfish-core/lib/assembly.hpp:479:        "xyz.openbmc_project.Association.Definitions", "Associations");
redfish-core/lib/log_services.hpp:4768:                if (property.first == "Associations")
```
upstream code:
```
-bash-4.2$ git grep -in "\"Associations\""
-bash-4.2$ 
```

I am not sure without use in the upstream we can add so let me know if we can add.  Now marking as resolved and let me know still you have concerns.